`Reusing typecheck result` session 2025-03-28 (#281)
amplifying-fsharp/amplifying-fsharp.github.io: src/sessions/2025-03-28.md

---

> title: "Reusing typecheck results between compiler runs"

Reviewer: 'Pickling' data in F#

> title: "Reusing typecheck results between compiler runs"

Reviewer: ```suggestion
title: Data 'pickling' in the F# compiler
```

> preview: "Reusing typecheck results"

Reviewer: 'Pickling' data in F#

isDraft: true
date: 2025-03-28T14:00:00.000Z
id: "2025/03/28"
champion: "Petr Semkin"
zoomLink: "https://us06web.zoom.us/j/88634727094?pwd=odJCovtHmlrbepkgXgw57MbiE9qPPu.1"
zoomPasscode: "typecheck"

> issueLink: "https://github.com/dotnet/fsharp/blob/main/docs/reusing-typechecking-results.md"

Reviewer: ```suggestion
issueLink: "https://github.com/dotnet/fsharp/blob/main/src/Compiler/TypedTree/TypedTreePickle.fs"
```

company: "Microsoft"
youtubeId: ""
---

## Topic

> In this session, we will discuss the work on [reusing typecheck results between compiler runs](https://github.com/dotnet/fsharp/blob/main/docs/reusing-typechecking-results.md)

Reviewer: ```suggestion
In this session, we will discuss data serialization (aka 'pickling') in the F# compiler - how F# code is stored on the disc. The talk will also touch on the topic of [reusing typecheck results between compiler runs](https://github.com/dotnet/fsharp/blob/main/docs/reusing-typechecking-results.md), the feature which is supposed to add logic there.
```
